test/libsolidity/SolidityParser.cpp

```diff
@@ -1047,6 +1047,61 @@ BOOST_AUTO_TEST_CASE(using_for)
 	BOOST_CHECK(successParse(text));
 }
 
+BOOST_AUTO_TEST_CASE(inline_array_declaration_lvalue)
+{
+	char const* text = R"(
+		contract c {
+			uint[] a;
+			function f() returns (uint) {
+				a = [1,2,3];
+				return (a[3]);
+			}
+		}
+	)";
+	BOOST_CHECK(successParse(text));
+}
+
+BOOST_AUTO_TEST_CASE(inline_array_declaration_self)
+{
+	char const* text = R"(
+		contract c {
+			uint[] a;
+			function f() returns (uint) {
+				return ([1,2,3][0]);
+			}
+		}
+	)";
+	BOOST_CHECK(successParse(text));
+}
+
+BOOST_AUTO_TEST_CASE(inline_array_empty_cells_check_beginning)
+{
+	char const* text = R"(
+		contract c {
+			uint[] a;
+			function f() returns (uint, uint) {
+				a = [,2,3];
+				return (a[3], [,3,4][0]);
+			}
+		}
+	)";
+	BOOST_CHECK(!successParse(text));
+}
+
+BOOST_AUTO_TEST_CASE(inline_array_empty_cells_check_commas)
+{
+	char const* text = R"(
+		contract c {
+			uint[] a;
+			function f() returns (uint, uint) {
+				a = [1, ,3];
```

Comment:
Perhaps make these negative checks a bit simpler. If they fail due to a different reason than the expected, we will not notice it.